src/high-level/types/complex-double-float.lisp

```diff
@@ -0,0 +1,492 @@
+          (size vector1) (size vector2))
+  (loop :for i :below (size vector1)
+        :sum (* (tref vector1 i) (conjugate (tref vector2 i)))))
+
+(defmethod orthonormalize!((m matrix/complex-double-float))
+  "Applies Gram-Schmidt to the columns of a full rank square matrix to produce a unitary matrix, replacing the elements"
+  (assert-square-matrix m)
+  ;; consider each column
+  (dotimes (j (ncols m))
+    ;; consider each preceding column, which together form an orthonormal set
+    (dotimes (jp j)
+      ;; compute the dot product of the columns...
+      (let ((scalar
+              (loop :for i :below (nrows m)
+                    :sum (* (tref m i j)
+                            (conjugate (tref m i jp))))))
+        ;; ... and do the subtraction.
+        (dotimes (i (nrows m))
+          (setf (tref m i j)
+                (cl:- (tref m i j)
+                      (* scalar
+                         (tref m i jp)))))))
+    ;; now j is orthogonal to the things that came before it. normalize it.
+    (let ((scalar
+            (sqrt
+             (loop :for i :below (nrows m)
+                   :sum (* (abs (tref m i j))
+                           (abs (tref m i j)))))))
+      (dotimes (i (nrows m))
+        (setf (tref m i j)
+              (/ (tref m i j) scalar)))))
+  m)
+
+(defmethod = ((tensor1 tensor/complex-double-float) (tensor2 tensor/complex-double-float) &optional (epsilon +double-comparison-threshold-strict+))
+  (unless (equal (shape tensor1) (shape tensor2))
+    (return-from = nil))
+  (map-indexes
+   (shape tensor1)
+   (lambda (&rest pos)
+     (unless (> epsilon
+                (abs (cl:- (apply #'tref tensor1 pos)
+                           (apply #'tref tensor2 pos))))
+       (return-from = nil))))
+  t)
+
+(defmethod = ((tensor1 matrix/complex-double-float) (tensor2 matrix/complex-double-float) &optional (epsilon +double-comparison-threshold-strict+))
+  (unless (equal (shape tensor1) (shape tensor2))
+    (return-from = nil))
+  (map-indexes
+   (shape tensor1)
+   (lambda (&rest pos)
+     (unless (> epsilon
+                (abs (cl:- (apply #'tref tensor1 pos)
+                           (apply #'tref tensor2 pos))))
+       (return-from = nil))))
+  t)
+
+
+(defmethod mult ((a matrix/complex-double-float) (b matrix/complex-double-float) &key target (alpha #C(1d0 0d0)) (beta #C (0d0 0d0)) transa transb)    
+  ;; (check-type target (or matrix/complex-double-float nil))
+  (check-type transa (member nil :n :t :c))
+  (check-type transb (member nil :n :t :c))
+  ;; TODO: make sure we can trans mixed order matrices (hard)
+  ;; Maybe just error with :row-major if :transX is :t or :c
+
+  ;; TODO: You can't just transpose them if they are the wrong order. Mixed order stuff causes issues. possibly needs to be reindexed
+  (let* ((transa (or transa
+                     (if (eq :row-major (order a)) :t :n)))
```

Comment:
I think you want `:c` here to get the conjugate transpose instead of the transpose.

The same thing applies everywhere you pass TRANS to BLAS with complex matrices/vectors.

Reply:
Also you should quote them before @stylewarning sees.

Reply:
This is to account for storing a matrix in row-major then trying to use fortran which is column major. I still haven't gotten it to not crash though